### ISSUE TITLE
a room listed in the directory should be accessible

### ISSUE THREAD

**Issue by AmandineLP** [vector-im/riot-web#3983]
trying to access #testingGuest:matrix.org as a guest from the directory gives the error message '#testingGuest:matrix.org does not exist'. Whatever is the reason for this msg to appear (deleted alias? not accessible to guests?) should be managed in a much nicer way: if something is missing the room should not be in the directory.

**Comment by AmandineLP** [comment]
ok I think the bottom line is that the address was removed => when removing the only address of a room there should be a warning saying "we will unlist it" and actually unlist it

I found this room again, added an address (#testingguestaccess:matrix.org) but not selected it in the drop down (the first added address should be selected by default in the drop down). I was able to join this address directly but still not from the directory.

I got back, selected the new address (#testingguestaccess:matrix.org) in the drop down and the wrong one (#testingGuest:matrix.org  was displayed in the browser.

**Comment by lampholder** [comment]
This is pretty smelly - there certainly shouldn't be items in the room directory that are 100% inaccessible to everyone.

@AmandineLP - is it possible that the alias was removed after you had loaded the directory but before you had tried to enter the room? Was the room still in the list after a refresh?

**Comment by AmandineLP** [comment]
Nope, I'm sure the alias was removed ages ago (it's a one of my test rooms and I suspect I did remove it and I haven't played with it since ages). Plus I've been always accessing the directory from a new incognito window for testing purposes.

**Comment by turt2live** [comment]
Related: https://github.com/vector-im/riot-web/issues/808

**Comment by richvdh** [comment]
> trying to access #testingGuest:matrix.org as a guest from the directory gives the error message '#testingGuest:matrix.org does not exist'.

I think this amounts to https://github.com/matrix-org/synapse/issues/6346.

**Comment by t3chguy** [comment]
Moved to https://github.com/matrix-org/synapse/issues/12733
